docs/reference-docs/auth-backend-and-frontend.md

```diff
@@ -0,0 +1,228 @@
+    TASKS_CREATE = '/orchestrator/processes/create/task', // called when determining if create if actions that trigger a task should be displayed
+    TASKS_RETRY_ALL = '/orchestrator/processes/all-tasks/retry', // called when determining if create if actions that trigger retry all tasks task should be displayed
+    SETTINGS_FLUSH_CACHE = '/orchestrator/settings/flush-cache', // called when determining if a button to flush cache should be displayed
+    SET_IN_SYNC = '/orchestrator/subscriptions/set-in-sync', // called when determining if a button to set a subscription in sync should be displayed
+}
+```
+
+Note: Components that are hidden for unauthorized users are still part of the frontend application, authorization just makes sure
+unauthorized users are not presented with actions they are not allowed to take. The calls these actions
+make can still be made through curl calls for example. Additional authorization needs to be implemented on these calls on the backend.
+
+### Backend
+
+**ENV variables**
+These variables need to be set for authentication to work on the backend
+
+```
+...
+# OIDC settings
+OAUTH2_ACTIVE: bool = True
+OAUTH2_AUTHORIZATION_ACTIVE: bool = True
+OAUTH2_RESOURCE_SERVER_ID: str = ""
+OAUTH2_RESOURCE_SERVER_SECRET: str = ""
+OAUTH2_TOKEN_URL: str = ""
+OIDC_BASE_URL: str = ""
+OIDC_CONF_URL: str = ""
+
+# OPtional OPA settings
+OPA_URL: str = ""
+```
+
+With the variables provided requests to endpoints will return 403 errorcodes for users that are not logged in and 401 error codes for users that are not authorized for a call or part of a call
+
+#### Customization
+
+When initiating the `OrchestratorCore` class it's [`auth_manager`][6] property is set to `AuthManager`. AuthManager is provided by [oauth2_lib][7].
+
+There are 3 methods from `AuthManager` that are used for authentication and authorization. `AuthManager` provides defaults and methods to override
+the set custom fuctions to override the default implementations. The default implementations are:
+
+`authentication`: Method that implements returning the OIDC user from the OIDC introspection endpoint
+
+`authorization`: Method that applies OPA decisions to HTTP requests for authorization. Uses OAUTH2 settings and request information to authorize actions.
+Sends this payload to the opa_url specified in OPA_URL setting to get a decision.
+
+```
+        opa_input = {
+            "input": {
+                **(self.opa_kwargs or {}),
+                **(user_info or {}),
+                "resource": request.url.path,
+                "method": request_method,
+                "arguments": {"path": request.path_params, "query": {**request.query_params}, "json": json},
+            }
+        }
+```
+
+Note:
+During app initialization a **is_bypassable_request** method can be passed into the app that receives the Request object
```

Comment:
here we can say for example for interacting with LSO and the callback step you can bypass the request since authentication happens in the callback step API itself and no need here.